### DISCUSSION TITLE
how to pass value data type to updateMyData for editable cell?

### DISCUSSION THREAD

**Discussion by stliang** [TanStack/table#2898] [Q&A]
From this example for editable cell:
https://github.com/tannerlinsley/react-table/blob/master/examples/editable-data/src/App.js

```
  // When our cell renderer calls updateMyData, we'll use
  // the rowIndex, columnId and new value to update the
  // original data
  const updateMyData = (rowIndex, columnId, value) => {
    // We also turn on the flag to not reset the page
    setSkipPageReset(true)
    setData(old =>
      old.map((row, index) => {
        if (index === rowIndex) {
          return {
            ...old[rowIndex],
            [columnId]: value,
          }
        }
        return row
      })
    )
  }
```

If the value is an Int, once a new value is added, the new value turn into a string instead of Int.  Is there a way to tell updateMyData what data type value is?

**Comment by JRRS1982** [comment]
Isn't updateMyData just a generic callback function - i think you can add whatever type you like to value - presumably the type of the value you are requesting in the get, it may be easier in typescript. 

What i am currently trying to understand is where old comes from - as setData is passed the new state for data, so... what value does old have? Its not declared anywhere so data is set undefined if i am not mistaken? Feel like it should be setData(value => etc
